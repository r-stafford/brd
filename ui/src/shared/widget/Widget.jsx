import React from 'react';
import Rnd from 'react-rnd';
import { keys } from 'lodash';

export default class Widget extends React.Component {

  static DEFAULT = {
    x: 0,
    y: 0,
    height: 100,
    width: 200
  };

  static Heading(props) {
    return (
      <div className="widget-heading">
        { props.children }
      </div>
    );
  }

  static Footer(props) {
    return (
      <div className="widget-footer">
        { props.children }
      </div>
    );
  }

  static Body(props) {
    return (
      <div className="widget-body">
        { props.children }
      </div>

    );
  }

  render() {
    const { props } = this;

    return (
      <Rnd 
        className={ `widget-container ${props.className || ''}` }
        bounds={ props.bounds || 'parent' }
        default={ props.default || Widget.DEFAULT }
        onResize={ props.onResize }
        onDragStop={ props.onDragStop }
        ref={ component => { this.rnd = component; }}> 
        { props.children }
      </Rnd>
    );
  }

}
