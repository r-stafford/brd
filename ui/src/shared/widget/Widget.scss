.widget-container {
  border: 1px solid #ddd;
  border-radius: 4px;
  margin: 3px 3px;
  padding: 0px 0px;
  overflow: hidden;

  .widget-heading { 
    font-size: 12px;
    font-weight: 400;
    padding: 2px 4px;
    background-color: #eee;
    height: 20px;
  }

  .widget-body {
    height: calc(100% - 40px);
  }

  .widget-footer {
    font-weight: 400;
    padding: 2px 4px;
    height: 20px;
    background-color: #eee;
  }
}