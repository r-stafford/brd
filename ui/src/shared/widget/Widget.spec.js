import React from 'react';
import ReactDOM from 'react-dom';
import Widget from './Widget';
import renderer from 'react-test-renderer';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

jest.mock('../../services/local-storage/LocalStorageService');

configure({ adapter: new Adapter() });
describe('Widget', () => {  

    it('Should render correctly', () => {
      const component = renderer.create(<Widget widget-id="mock-widget"/>);
      expect(component.toJSON()).toMatchSnapshot();

      const instance = component.getInstance()
      expect(instance.state).toMatchSnapshot('initial state');     
    });

    it('always renders a Heading', () => {
      const instance = Widget.Heading({ children:[] });
      expect(instance.state).toMatchSnapshot('initial state');     
    });

    it('always renders a Footer', () => {
      const instance = Widget.Footer({ children:[] });
      expect(instance.state).toMatchSnapshot('initial state');     
    });

    it('always renders a Body', () => {
      const instance = Widget.Body({ children:[] });
      expect(instance.state).toMatchSnapshot('initial state');     
    });
});