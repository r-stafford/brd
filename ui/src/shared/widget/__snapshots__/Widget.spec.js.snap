// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget Should render correctly 1`] = `<div />`;

exports[`Widget Should render correctly: initial state 1`] = `null`;

exports[`Widget always renders a Body: initial state 1`] = `undefined`;

exports[`Widget always renders a Footer: initial state 1`] = `undefined`;

exports[`Widget always renders a Heading: initial state 1`] = `undefined`;
